### PR TITLE
Store shared camera ptr for reconstruction images

### PR DESCRIPTION
**Pull request by ahojnnes**

### CODE REVIEW

File: src/colmap/scene/reconstruction.cc [hunk 1]
```diff
@@ -117,6 +117,13 @@ void Reconstruction::AddCamera(struct Camera camera) {
 }
 
 void Reconstruction::AddImage(class Image image) {
+  THROW_CHECK(image.HasCameraId());
+  const auto camera_it = cameras_.find(image.CameraId());
```
Comment (B1ueber2y) [inline src/colmap/scene/reconstruction.cc:121]: Should we throw assertion error here on that the camera must exist

File: src/colmap/scene/reconstruction.h [hunk 2]
```diff
@@ -102,7 +102,9 @@ class Reconstruction {
   // might be taken by the same camera.
   void AddCamera(struct Camera camera);
 
-  // Add new image.
```
Comment (B1ueber2y) [inline src/colmap/scene/reconstruction.h:105]: We should also add comments at the two reconstruction IO functions like: https://github.com/colmap/colmap/pull/2702/files#diff-6ff1c633d8187e2cb2de82d2b9217f65046013c4edfd3c8eaf53e0a95b91461cR40.

**Comment by B1ueber2y** [comment]
This looks great to me. Should we also add the ProjectPoint function we had in the other PR before?

File: src/colmap/scene/reconstruction.cc (hunk 1 above)
Comment (ahojnnes) [reply src/colmap/scene/reconstruction.cc]: Done.

File: src/colmap/scene/reconstruction.h (hunk 2 above)
Comment (ahojnnes) [reply src/colmap/scene/reconstruction.h:105]: Done

**Comment by ahojnnes** [comment]
> This looks great to me. Should we also add the ProjectPoint function we had in the other PR before?

It's already there.

**Review by B1ueber2y — APPROVED** [review]

**Review by sarlinpe — APPROVED** [review]

**Pull request merged**